introduce the nucleo_h7s3l78 nucleo144 board from STMicroelectronics

FRASTM opened the pull request:
Add the nucleo_h7s3l8 board from STMicroelectronics 
This Nucleo 144 board has **stm32h7S3L8** mcu.

Refer to the https://www.st.com/en/evaluation-tools/nucleo-h7s3l8.html


$ west build -p auto -b nucleo_h7s3l8 samples/hello_world/
```
*** Booting Zephyr OS build v4.0.0-rc3-49-g7500cdb33f48 ***                     
Hello World! nucleo_h7s3l8/stm32h7s3xx         
```

FRASTM committed:
boards: st: nucleo_h7s3l78 nucleo144 board

Add the nucleo_h7s3l8 board from STMicroelectronics
This Nucleo 144 board has stm32h7s3L8 mcu.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

nordicjm commented on boards/st/nucleo_h7s3l8/doc/img/nucleo_h7s3l8.jpg:
trim space around image, (optionally) make background transparent, save as webp, put webp through https://tinypng.com/

nordicjm replied:
Seems unresolved?

nordicjm commented on boards/st/nucleo_h7s3l8/doc/index.rst:
```diff
@@ -0,0 +1,324 @@
+Supported Features
+==================
+
+The Zephyr nucleo_h7s3l8 board configuration supports the following hardware
```
```suggestion
The Zephyr ``nucleo_h7s3l8`` board target supports the following hardware
```

nordicjm commented on boards/st/nucleo_h7s3l8/nucleo_h7s3l8_defconfig:
```diff
@@ -0,0 +1,20 @@
+# Enable HW stack protection
+CONFIG_HW_STACK_PROTECTION=y
+
+# enable uart driver
+CONFIG_SERIAL=y
+# enable console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# enable GPIO
+CONFIG_GPIO=y
```
```suggestion
# Enable UART driver
CONFIG_SERIAL=y

# Enable console
CONFIG_CONSOLE=y
CONFIG_UART_CONSOLE=y

# Enable GPIO
CONFIG_GPIO=y
```

FRASTM committed:
tests: drivers: adc api testing on the stm32h7s3l8 nucleo

Add the overlay file to run the tests/drivers/adc/adc_api
on the nucleo_h7s3l8

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
add comments from @nordicjm 
Fix CI error on wdg_api and adc_api

FRASTM committed:
tests: drivers: watchdog api testing on the stm32h7rs boards

Add the overlay file to run tests/drivers/watchdog/wdt_basic_api
on the nucleo_h7s3l8 and stm32h7s78_dk target boards

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented on tests/drivers/adc/adc_api/boards/nucleo_h7s3l8.overlay:
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics4
```
done

FRASTM commented on tests/drivers/adc/adc_api/boards/nucleo_h7s3l8.overlay:
```diff
@@ -0,0 +1,27 @@
+
+/ {
+	zephyr,user {
+		/* adjust channel number according to pinmux in board.dts */
```
done

FRASTM commented on tests/drivers/watchdog/wdt_basic_api/testcase.yaml:
```diff
@@ -45,6 +45,8 @@ tests:
     filter: dt_compat_enabled("st,stm32-window-watchdog") or dt_compat_enabled("st,stm32-watchdog")
     extra_args: DTC_OVERLAY_FILE="boards/stm32_wwdg_h7.overlay"
```
corrected

FRASTM commented:
reducing the image size with nucleo_h7s3l8.webp  file

bjarki-andreasen commented on tests/drivers/watchdog/wdt_basic_api/testcase.yaml:
```diff
@@ -45,6 +45,8 @@ tests:
     extra_args: DTC_OVERLAY_FILE="boards/stm32_wwdg_h7.overlay"
     platform_allow:
+      - nucleo_h7s3l8
+      - stm32h7s78_dk
```
Adding `stm32h7s78_dk` to this test suite is not necessarily part of introducing the `nucleo_h7s3l78` but it is very minor so I think its fine :)
